services/basic_types.proto

```diff
@@ -79,9 +79,31 @@ message AccountID {
     int64 realmNum = 2;
 
     /**
-     * A nonnegative account number unique within its realm
+     * The account number unique within its realm which can be either a non-negative integer or an alias public key.
+     * For any AccountID fields in the query response, transaction record or transaction receipt only accountNum will
+     * be populated.
      */
-    int64 accountNum = 3;
+    oneof account {
+
+        /**
+         * A non-negative account number unique within its realm
+         */
+        int64 accountNum = 3;
+
+        /**
+         * A public key to be used as the account's alias. Only a primitive key is supported as an alias Key
+         * (ThresholdKey and KeyList are not supported).
```

Comment:
We will need to be sure to test this explicitly in unit and integration tests.

May be worth enumerating the keys as "this currently only includes [list of key fields]" or add contractID and delegatable_contract_id to the list in parentheses.

Reply:
@shemnon  Added unit tests in the test plan to verify this.